test/Dialect/Seq/hw-memsim.mlir

```diff
@@ -2,14 +2,17 @@
 // RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{read-enable-mode=ignore})" %s | FileCheck %s --check-prefixes=COMMON,IGNORE
 // RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{read-enable-mode=zero})" %s | FileCheck %s --check-prefixes=COMMON,ZERO
 // RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{add-mux-pragmas})" %s | FileCheck %s --check-prefixes=COMMON,PRAMGAS
-// RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{disable-mem-randomization})" %s | FileCheck %s --check-prefix COMMON --implicit-check-not RANDOMIZE_MEM
-// RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{disable-reg-randomization})" %s | FileCheck %s --check-prefix COMMON --implicit-check-not RANDOMIZE_REG
-// RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{disable-mem-randomization disable-reg-randomization})" %s | FileCheck %s --check-prefix COMMON --implicit-check-not RANDOMIZE_REG --implicit-check-not RANDOMIZE_MEM
+// RUN: circt-opt -pass-pipeline="builtin.module(hw-memory-sim{disable-mem-randomization})" %s | FileCheck %s --check-prefix COMMON --check-prefix NO-RAND-MEM
```

Comment:
The `-NOT`'s here are applied to the entire output (only directive on each run) which makes sinking them to line 80 a bit unexpected -- that seems to suggest the checks start at that point or are relevant there.

What's the thinking here -- these FileCheck lines are doing too much, let's put them into a series of directives under one prefix each instead?

I think this change is functionally equivalent to putting the `sv.ifdef.procedural @RANDOMIZE_REG` in the FileCheck implicit-check-not -- is that right? But that makes an already long line longer.

Anyway other than the new prefixes' file-wide check-not being defined unexpectedly late, this is mostly preference/stylistic.

( Looks like `NO-RAND` could be `--check-prefixes NO-RAND-MEM,NO-RAND-REG` ? )